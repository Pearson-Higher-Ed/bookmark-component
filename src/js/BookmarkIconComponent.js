import React from 'react';
import { IntlProvider } from 'react-intl';
import InternationalSupport from './InternationalSupport';
import BookmarkIcon from './BookmarkIcon';

export const BookmarkIconComponent = function BookmarkIconComponent(paramsObj) { // eslint-disable-line import/prefer-default-export
  const intlObj = new InternationalSupport(paramsObj.locale);
  return (<IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}>
    <BookmarkIcon
    	books={paramsObj.books}
      onBookClick={paramsObj.onBookClick}
    />
  </IntlProvider>);
};
