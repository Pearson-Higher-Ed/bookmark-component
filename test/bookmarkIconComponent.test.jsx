/**
PEARSON PROPRIETARY AND CONFIDENTIAL INFORMATION SUBJECT TO NDA
 *  Copyright Â© 2017 Pearson Education, Inc.
 *  All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Pearson Education, Inc.  The intellectual and technical concepts contained
 * herein are proprietary to Pearson Education, Inc. and may be covered by U.S. and Foreign Patents,
 * patent applications, and are protected by trade secret or copyright law.
 * Dissemination of this information, reproduction of this material, and copying or distribution of this software
 * is strictly forbidden unless prior written permission is obtained
 * from Pearson Education, Inc.
**/
/* eslint-disable */


import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { IntlProvider } from 'react-intl';
import { InternationalSupport } from '@pearson-incubator/aquila-js-core';
import BookmarkIconComponent from '../src/js/BookmarkIconComponent';
import msgObject from '../translations';

const isCurrentPageBookmarked = {
  isCurrentPageBookmarked: function() {
    return true
  }
}

const intlObj = new InternationalSupport(msgObject, 'en');
const data = {
  "BookmarkIconFilled": "Bookmarked",
  "BookmarkIconUnfilled": "Not Bookmarked"
}

it('renders the component correctly', () => {
  const tree = renderer
    .create(
      <MuiThemeProvider>
        <IntlProvider 
        locale={intlObj.getLocale()} 
        messages={intlObj.getMessages()}>
          <BookmarkIconComponent data={isCurrentPageBookmarked} />
        </IntlProvider>
      </MuiThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});




