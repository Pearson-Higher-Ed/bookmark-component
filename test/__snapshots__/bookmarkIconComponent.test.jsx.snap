// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<div>
  <div
    aria-label="Bookmarked"
    className="filled"
  >
    <svg
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": "20px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "16px",
        }
      }
      tabIndex="0"
      viewBox="0 0 15 18"
    >
      <path
        d="M11.5,-0.0004 L0.5,-0.0004 C0.224,-0.0004 0,0.2236 0,0.4996 L0,16.4996 C0,16.6996 0.119,16.8806 0.302, 16.9586 C0.488,17.0396 0.699,16.9996 0.844,16.8636 L6,11.9876 L11.156,16.8636 C11.251,16.9526 11.375, 16.9996 11.5,16.9996 C11.566,16.9996 11.634,16.9866 11.698,16.9586 C11.881,16.8806 12,16.6996 12, 16.4996 L12,0.4996 C12,0.2236 11.776,-0.0004 11.5,-0.0004 L11.5,-0.0004 Z"
        fill="#FFF"
      />
    </svg>
  </div>
</div>
`;
